scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -232,8 +232,7 @@ object Execution {
               try {
                 ExecutionContext.newContext(conf)(fd, mode).run
               } catch {
-                // Try our best to complete the future
-                case e: Throwable => Future.failed(e)
+                case NonFatal(e) => Future.failed(e)
```

Comment:
I think we should be catching this and possibly re-throw after we satisfiy the future. Your inclination i think i right in that will manifest poorly otherwise with hung programs with no visible reason why.